website/docs/guides/customize-schema-alias.md

```diff
@@ -0,0 +1,473 @@
+
+
+```jinja2
+
+{% macro generate_schema_name(custom_schema_name=none, node=none) -%}
+
+    {%- set default_schema = target.schema -%}
+    
+    {# If the CI Job does not exist in it's own environment, use the target.name variable inside the job instead #}
+    {# {%- if target.name == 'CI' -%} #} 
+    
+    {%- if env_var('DBT_ENV_TYPE','DEV') == 'CI' -%}
+        
+        ci_schema
+        
+    {%- elif custom_schema_name is none -%}
+        
+        {{ default_schema }}
+    
+    {%- else -%}
+        
+        {{ default_schema }}_{{ custom_schema_name | trim }}
+    
+    {%- endif -%}    
+
+{%- endmacro %}
+
+```
+
+```jinja2
+
+{% macro generate_alias_name(custom_alias_name=none, node=none) -%}
+
+    {# If the CI Job does not exist in it's own environment, use the target.name variable inside the job instead #}
+    {# {%- if target.name == 'CI' -%} #}   
+    {%- if  env_var('DBT_ENV_TYPE','DEV') == 'CI' -%}
+
+        {%- if custom_alias_name -%}
+
+            {{ target.schema }}__{{ node.config.schema }}__{{ custom_alias_name | trim }}
+
+        {%- elif node.version -%}
+
+            {{ target.schema }}__{{ node.config.schema }}__{{ node.name ~ "_v" ~ (node.version | replace(".", "_")) }}
+
+        {%- else -%}
+
+            {{ target.schema }}__{{ node.config.schema }}__{{ node.name }}
+
+        {%- endif -%}
+    
+    {%- else -%}
+
+        {%- if custom_alias_name -%}
+
+            {{ custom_alias_name | trim }}
+
+        {%- elif node.version -%}
+
+            {{ return(node.name ~ "_v" ~ (node.version | replace(".", "_"))) }}
+
+        {%- else -%}
+
+            {{ node.name }}
+
+        {%- endif -%}
+
+    {%- endif -%}
+
+{%- endmacro %}
+
+```
+
+This will generate the following outputs for a model called `my_model` with a custom schema of `marketing`, preventing any overlap of objects between dbt runs from different contexts.
+
+
+| Context     |Target database| Target schema | Resulting object                          |
+|-------------|:-------------:|:-------------:|:----------------------------------------: |
+| Developer 1 | dev           | dbt_dev1      |dev.dbt_dev1_marketing.my_model            |
+| Developer 2 | dev           | dbt_dev2      |dev.dbt_dev2_marketing.my_model            |
+| CI PR 123   | ci            | dbt_pr_123    |ci.ci_schema.dbt_pr_123_marketing_my_model |
+| CI PR 234   | ci            | dbt_pr_234    |ci.ci_schema.dbt_pr_234_marketing_my_model |
+| Production  | prod          | analytics     |prod.marketing.my_model                    |
+
+
+### What NOT to do
+
+
+### Update generate_schema_name() to always use the custom schema if it is set
+
+
+Some people prefer to only use the custom schema when it is set instead of concatenating the default schema with the custom one, as it happens in the out of the box behavior.
+
+
+#### Problem
+
+When modifying the default macro for `generate_schema_name()`, this might result iney might creatinge this new version.
```

Comment:
there are some typos here, suggest updating them. didn't know what the 'iney' meant